connect/connect.go

```diff
@@ -0,0 +1,61 @@
+// Package connect provides a Connector which allows to connect method
+// definition with their implementations in plugins
+package connect
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/nextmv-io/sdk/plugin"
+)
+
+// Connector connects methods with their implementations in plugins.
+type Connector struct {
+	connected    map[any]struct{}
+	mtx          sync.Mutex
+	slug, prefix string
+}
+
+// NewConnector creates a new Connector.
+func NewConnector(slug, prefix string) *Connector {
+	return &Connector{
+		connected: make(map[any]struct{}),
+		slug:      slug,
+		prefix:    prefix,
+	}
+}
+
+// Connect connects a method with its implementation.
+func Connect[T any](c *Connector, target *T, suffix ...string) {
+	if _, ok := c.connected[target]; ok {
+		return
+	}
+
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
+	if _, ok := c.connected[target]; ok {
+		return
+	}
```

Comment:
Do we need this twice?

Reply:
Yes, the first call is a performance optimization. Then you lock. When the lock is released the second waiting goroutine, when it passes the lock, doesn't need to do all the code after that check again. You could do all this without a lock. It was like that and is fine from my point of view, so I left it that way.